src/main/java/uk/gov/justice/hmpps/prison/service/reference/OffenceService.java

```diff
@@ -192,4 +194,14 @@ public void unlinkOffencesFromSchedules(final List<OffenceToScheduleMappingDto>
             offenceIndicatorRepository.deleteByIndicatorCodeAndOffence_Code(o.getSchedule().getCode(), o.getOffenceCode())
         );
     }
+
+    @Transactional
+    public void updateOffenceActiveFlag(final OffenceActivationDto offenceActivationDto) {
+        final var offence = offenceRepository.findById(new PK(offenceActivationDto.getOffenceCode(), offenceActivationDto.getStatuteCode())).orElseThrow(
+            EntityNotFoundException.withMessage("The offence with code %s doesnt exist", offenceActivationDto.getOffenceCode())
```

Comment:
You probably need to mention the statute code in the message too

Reply:
Thanks - has been changed